src/clp_ffi_js/ir/UnstructuredIrStreamReader.cpp

```diff
@@ -187,9 +186,13 @@ auto UnstructuredIrStreamReader::decode_range(size_t begin_idx, size_t end_idx,
         } else {
             log_event_idx = i;
         }
-        auto const& log_event{m_encoded_log_events[log_event_idx]};
+        auto const& log_event_with_filter_data{m_encoded_log_events[log_event_idx]};
+        auto const& log_level = log_event_with_filter_data.get_log_level();
+        auto const& timestamp = log_event_with_filter_data.get_timestamp();
 
-        auto const parsed{log_event.get_message().decode_and_unparse()};
+        auto const& unstructured_log_event = log_event_with_filter_data.get_log_event();
```

Comment:
How about inserting this above line 190 so that we're essentially pulling out all the fields of the `LogEventWithFilterData`?